### DISCUSSION TITLE
One to Many Relationships - not posting to DB

### DISCUSSION THREAD

**Discussion by Hmichas** [noam-honig/radweb#423] [Q&A]
Good Afternoon!

I'm very new at this and this is my first time ever diving into the discussions on Git, so I would like to apologize in advance if what I'm saying is redundant or a stupid question.

Right now I'm working on a web application using Next.js, and my team and I decided to use Remult for our CRUD and DB operations.
It's been working great so far but I cannot for the life of me get any entity relationships other than Many to One to actually work.
I feel like I've read over the documentation a thousand times and I still don't understand why it isn't working.

We're using Next.js 14, with the App Router, and hosting it on Vercel using a Postgres DB

This is a simplified version of what I'm working on:

artist.ts
```typescript
import { Entity, Fields, Relations, remult } from 'remult';
import { Artwork } from './artwork';

@Entity('artist', { allowApiCrud: true })
export class Artist 
{
	@Fields.cuid()
	id!: string:
	
	@Fields.string()
        name!: string;
	
	@Fields.string()
        bio?: string
	
	@Relations.toMany(() => Artwork)
        artworks?: Artwork[];
}
// I've also tried artworks?: Artwork[] = [];, still didn't work
```

artwork.ts
```typescript
import { Entity, Fields, Relations, remult } from 'remult';
import { Artist } from './artist';

@Entity('artwork', { allowApiCrud: true })
export class Artwork
{
	@Fields.cuid()
	id!: string:
	
	@Fields.string()
        title!: string;

        @Relations.toOne(() => Artist)
        artist!: Artist;
}
```

So the `@Relations.toOne(() => Artist)` is working, and visible on the admin UI, but the column for 'artworks' isn't even showing up on the Artist entity at all.
I've tried all of these different variations of calls to now success:
`@Relations.toMany<Artist, Artwork>(() => Artwork)`
`@Relations.toMany<Artist, Artwork>(() => Artwork, {defaultIncluded:true} )`
`@Relations.toMany(() => Artwork, 'artist' )`
`@Relations.toMany(() => Artwork, 'artwork' )`

Sorry in advance if this is stupid but I'm very lost and my deadline for deployment is quickly approaching and I have no clue how to fix this. 

Thanks!

edit: Fixed the entity naming typo and id fields, accidents caused from my lack of experience using markup lol, great catch!

**Comment by noam-honig** [comment]
Hi @Hmichas, In your example, both entities are called `Artist`.
Also - the `id` fields should be string (a `cuid` is a string id)

I've created a code sandbox with a code sample based on your entities - with the fixes above and it seems to work:
https://codesandbox.io/p/devbox/solitary-forest-6gh3c5?file=%2Fsrc%2Findex.ts%3A14%2C1

check it out and let me know what you think

**Reply by Hmichas** [reply]
Thank you for your reply!

I replicated the entity exactly like the ones in the sandbox and It's still not showing the Relationship on the admin panel :(
I'm going to try running a script to pull the relationship to the console, like in your demo, but as of right now my Artist data table still doesn't have a column for Artworks[ ].
I'm also hopefully lost on Many to Many relationships, but I figured that will have to wait until I could get the other half of that working...

**Comment by noam-honig** [comment]
Hi @Hmichas make sure you're using an up-to-date version of remult.

If you want -  I'll be happy to zoom in and see if I can help you track down the issue, you can find me on discord at:
https://discord.gg/GXHk7ZfuG5

Or simply email me at noam.honig@gmail.com and I'll send you a zoom link

**Reply by Hmichas** [reply]
That would be amazing, I'll email you with more details :)

**Comment by noam-honig** [comment]
I was playing around with stackblitz - so here's another example - now full-stack :)

https://stackblitz.com/edit/github-9cbhgm?file=src%2FApp.tsx

**Reply by Hmichas** [reply]
This is super helpful, thank you!!
